src/sds_data_model/dataframe.py

```diff
@@ -0,0 +1,254 @@
+                                       )
+        >>> wrapped_csv = check_csv = DataFrameWrapper.from_files(
+                                        name = "aes30",
+                                        data_path="dbfs:/mnt/lab/unrestricted/james.duffy@defra.gov.uk/aes30_in_aonbs.csv", 
+                                        read_file_kwargs = {'header' :True}
+                                       )
+        Args:
+            cls (_DataFrameWrapper): DataFrameWrapper class , defined above
+            data_path (str): path to data, 
+            metadata_path (Optional[str], optional): _description_. Defaults to None.
+            name (Optional[str], optional): Name for data, either supplied by caller or obtained from metadata title. Defaults to None.
+            read_file_kwargs (Optional[Dict[str,Any]], optional): Additional kwargs supplied by the caller, dependent on the function called. Defaults to None.
+        Returns:
+            _DataFrameWrapper: _description_
+        """
+        #_spark = spark if spark else sparksession.getActiveSession  .. operator
+        #need to replace spark with _spark
+        _spark = spark if spark else SparkSession.getActiveSession()
+
+        if read_file_kwargs:
+            read_file_kwargs = read_file_kwargs
+        else:
+            read_file_kwargs = {}
+
+        file_reader_pandas = {
+            ".xlsx": read_excel,
+            ".xls": read_excel,
+            ".xlsm": read_excel,
+            ".xlsb": read_excel,
+            ".odf": read_excel,
+            ".ods": read_excel,
+            ".odt": read_excel,
+        }
+
+        file_reader_spark = {
+            ".csv": _spark.read.csv,
+            ".json": _spark.read.json,
+            ".parquet": _spark.read.parquet,
+        }
+
+        suffix_data_path = Path(data_path).suffix
+
+        if suffix_data_path in file_reader_pandas.keys():
+            data = file_reader_pandas[suffix_data_path](data_path, **read_file_kwargs)
+            data = data.to_spark()
+
+        elif suffix_data_path in file_reader_spark.keys():
+            data = file_reader_spark[suffix_data_path](data_path, **read_file_kwargs)
+        elif suffix_data_path == ".gpkg":
+            data = read_gpkg(data_path, **read_file_kwargs)
+        else:
+            data = read_vector_files(
+                data_path,
+                **read_file_kwargs
+            )
+                  
+        metadata = _get_metadata(
+            data_path=data_path,  
+            metadata_path=metadata_path)
+        
+        _name = _get_name(
+            name=name,
+            metadata=metadata,
+        )
+                
+        return cls(
+            name=_name,
+            data=data,
+            metadata=metadata
+        )
+
+    def call_method(
+        self: Self, 
+        method_name: str,
+        /,
+        *args,
+        **kwargs,
+    ) -> Optional[Self]:
```

Comment:
It looks like `mypy` still isn't down with `Self` so this will need to be:
```suggestion
    def call_method(
        self: _DataFrameWrapper,
        method_name: str,
        /,
        *args,
        **kwargs,
    ) -> Optional[_DataFrameWrapper]:
```